wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -90,17 +91,28 @@ getHistory mCWalId mAccountId mAddrId = do
             accIds' <- getWalletAccountIds cWalId'
             pure (cWalId', accIds')
         (Nothing, Just accId)   -> pure (aiWId accId, [accId])
-    accAddrs <- map cwamId <$> concatMapM (getAccountAddrsOrThrow Ever) accIds
-    addrs <- case mAddrId of
-        Nothing -> pure accAddrs
-        Just addr ->
-            if addr `elem` accAddrs then pure [addr] else throwM errorBadAddress
-    first (filter (fits $ S.fromList addrs)) <$> getFullWalletHistory cWalId
+    allAccIds <- getWalletAccountIds cWalId
+
+    let getAccAddrs = map cwamId <$> concatMapM (getAccountAddrsOrThrow Ever) accIds
+        noAccFiltering = accIds == allAccIds
+
+    filterFunction <- case mAddrId of
+        Nothing -> if noAccFiltering
+            then pure identity
+            else getAccAddrs >>= pure . filterByAddrs
```

Comment:
Lil

Reply:
Lol, recalling the time when I wrote similar meme code

Reply:
Blya, nevyvoz, only now ponel where's memes